Fix `crypto_keyfile.bin` creation condition

vlinkz opened the pull request:
Moves `crypto_keyfile.bin` to `/boot`, and only creates the file if `/boot` is a partition and is encrypted or `/boot` is not a partition and `/` is encrypted.

Fixes #41
Closes #42

Todo:
- [x] test generated iso

vlinkz committed:
fix crypto_keyfile creation condition

emilazy reviewed (approved):
This looks correct to me but hopefully @ElvishJerricco (and @dalto8?) can take a look before merging, as I don’t have the cycles to do a bunch of VM testing myself right now.

dalto8 commented:
I did review this.  It looks right to me but I also didn't test it.

ElvishJerricco commented:
Hm.

![image](https://github.com/user-attachments/assets/80ba3c3d-cf8d-45db-8217-b9d7729936eb)

I just did the normal automated encrypted install with swap ("with hibernate"), no manual partitioning.

```
 dd: failed to open '/tmp/calamares-root-0ctojcd8/boot/crypto_keyfile.bin': No such file or directory
```

I'm guessing the `/boot` directory hasn't been created yet.

ElvishJerricco committed:
showNotEncryptedBootMessage: false

This is no longer needed, now that we correctly detect whether a
crypto_keyfile.bin is required in `/boot`.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/security-advisory-graphical-installer-disk-encryption-bug-regression-on-legacy-bios-systems-using-manual-partitioning-ghsa-vfxf-gpmj-2p25/50665/1
